modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -272,3 +278,14 @@ export function validateRawTransaction(rawTransaction: string): void {
     throw new ParseTransactionError('Invalid raw transaction');
   }
 }
+
+/**
+ * Validates staking address exists and is a valid Solana public key
+ *
+ * @param {string} stakingAddress The staking address to be validated
+ */
+export function validateStakingAddress(stakingAddress: string): void {
+  if (!stakingAddress || !isValidPublicKey(stakingAddress)) {
+    throw new BuildTransactionError('Invalid or missing stakingAddress, got: ' + stakingAddress);
+  }
+}
```

Comment:
can we do `validateAddress` instead to make this more generic?
we probably have this same check in every func that deals with addresses

Reply:
We did this because the error message should be clear in which address is missing or invalid.

Reply:
Maybe we can do something like this

```suggestion
/**
 * Validate the string address
 *
 * @param {string} address address to be validated
 * @param {string} parameterName name of the parameter in order to know which is the invalid address
 */
export function validateAddress(address: string, parameterName: string): void {
  if (!stakingAddress || !isValidPublicKey(stakingAddress)) {
    throw new BuildTransactionError(`Invalid or missing ${parameterName}, got: ${address}`);
  }
}
```

I can do it on my next PR about the Withdraw Builder  or I can create a refactor ticket since it touch all the builders.
What are your thoughts

Reply:
OK to do as a follow up PR.